src/js/libcs/OpSound.js

```diff
@@ -136,22 +169,23 @@ evaluator.playsin$1 = function(args, modifs) {
         }
     }
 
-    function setPhaseShift(oscNode) {
+    function setPhaseShift(oscNode, curphaseshift) {
```

Comment:
for simplicity, why not passing `phaseshift` instead of `phaseshift[0]==curphaseshift`? (or even using the variable, which is in the scope)

Reply:
for `precalculated == false` setPhaseShift is called for each harmonic individually to call createWave for each Node, so then we pass phaseshift[i] as curphaseshift into the function. otherwise we don't use curphaseshift but the phaseshift from the variablescope. It all gets a little twisted because we use multiple nodes for harmonics in the one case but a single node for the other case.